api/utils.py

```diff
@@ -617,3 +510,524 @@ def validate_query_string(
             if not key.startswith("tag.") and not key.startswith("tag_exists."):
                 return False
     return True
+
+
+@tracer.capture_method(capture_response=False)
+def serialise_neptune_obj(obj: dict, key_prefix: str = "") -> dict:
+    """Return a new dict with properties of type dict/list serialised into string"""
+    return {
+        f"{key_prefix}{constants.SERIALISE_PREFIX if isinstance(v, (list, dict)) else ''}{k}": (
+            json.dumps(v) if isinstance(v, (list, dict)) else v
+        )
+        for k, v in obj.items()
+    }
```

Comment:
```suggestion
    serialised = {}
    for k, v in obj.items():
        if isinstance(v, (list, dict)):
            serialised[f"{constants.SERIALISE_PREFIX}{k}"] = json.dumps(v)
        else:
            serialised[k] = v
    return serialised
```

A suggestion for readability, I found the logic nested in f-strings to be tricky to read.. I think particularly with the suggestion that follows this makes the logic more obvious.